webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx

```diff
@@ -430,6 +405,50 @@ export const GlideTable: React.FC<GlideTableProps> = ({
                 setMenuIsOpen(false);
               },
             },
+        ...(BANNED_SORT_COLUMNS.has(column.column)
+          ? []
+          : [{ type: 'divider' as const }, ...sortMenuItemsForColumn(column, sorts, onSortChange)]),
+        ...(BANNED_FILTER_COLUMNS.includes(column.column)
+          ? []
+          : [
+              { type: 'divider' as const },
+              {
+                icon: <Icon decorative name="filter" />,
+                key: 'filter',
+                label: 'Add Filter',
+                onClick: () => {
+                  setTimeout(() => {
+                    filterMenuItemsForColumn();
+                  }, 5);
```

Comment:
nit: same note about the timeout

Reply:
alternatively, I think it could at least be simplified to `setTimeout(filterMenuItemsForColumn, 5)`

Reply:
Sure. But I'm not exactly sure why the `setTimeout` is needed..

Reply:
I'm not sure either, though my assumption is that there was a race condition somewhere at some point.